src/main/java/com/example/weatherwaveapi/config/GeneralSettings.java

```diff
@@ -0,0 +1,44 @@
+package com.example.weatherwaveapi.config;
+
+import lombok.Data;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@Data
+@ConfigurationProperties(prefix = "application.settings")
+public class GeneralSettings {
```

Comment:
Let's still move away from creating one large class with settings and instead create separate classes for each API. Let's say at this moment we should have two classes YandexProperties and OpenWeatherProperties. Each will only have settings for their client. This will give us flexibility when adding new code. We will not need to change a large file, but make changes only in a small one. Also, when adding a new API, we will simply need to create a new class with properties that will manage this new API, and not make changes to the existing one.

Reply:
Ok, if you think this approach will be better, I agree. For my side, both approaches have pros and cons, I just prefer using one GeneralSettings class with nested classes/records, because it allows us to create only one instance of settings in services